pkg/render/logstorage.go

```diff
@@ -1419,3 +1395,60 @@ func (es elasticsearchComponent) kibanaPodSecurityPolicy() *policyv1beta1.PodSec
 	psp.GetObjectMeta().SetName("tigera-kibana")
 	return psp
 }
+
+// If user has set ResourceRequirements, replace individual field's default value with user's value.
+// If user provided both Limits and Requests, use them.
+// As ResourceRequirements.Limits must be >= ResourceRequirements.Requests,
+// if user provided just Limits, and Limits is <= default Requests, set Requests value as user's Limits value,
+// if user provided just Requests, and Requests is >= default Limits, set Limits value as user's Requests value.
+func overrideResourceRequirements(defaultReq *corev1.ResourceRequirements, userOverrides corev1.ResourceRequirements) {
+	if userOverrides.Limits.Cpu().Value() != 0 {
```

Comment:
I think we should check if the values was actually set, not that it's 0. If a user wants to set it to 0, they should be allowed. It may be that ends up being a special value in the future, but regardless we should not override user intent.